packages/celer-web-app/src/ui/frames/DocFrame.tsx

```diff
@@ -80,8 +89,30 @@ export const DocFrame: React.FC<DocFrameProps> = ({docLines})=>{
 				}
 			});
 			return [components, refs];
-
 		}
+		if (EnhancedScrollTrackerEnabled) {
```

Comment:
Why is this code block here? It seems exactly identical to the one above?

Reply:
The gate logic in this file is getting too crazy. Let's only keep your version (`ScrollProgressTrackerEnabled`) and remove the other ones